File: web/src/types/backup-run.ts
import type { BackupFile } from './backup-file';

export type BackupRunStatus = 'pending' | 'running' | 'completed' | 'success' | 'failed';

export interface BackupRun {
  id: number;
  backup_profile_id: number;
  start_time?: string;
  end_time?: string;
  status: BackupRunStatus;
  local_backup_path?: string;
  total_files?: number;
  total_size_bytes?: number;
  error_message?: string;
  log?: string;
  backup_files?: BackupFile[];
}

File: server/entity/backup_run.go
package entity

import "time"

// BackupRun represents each execution of a backup profile
type BackupRun struct {
	ID              uint      `gorm:"primaryKey" json:"id"`
	BackupProfileID uint      `gorm:"not null" json:"backup_profile_id"`
	StartTime       time.Time `json:"start_time"`
	EndTime         time.Time `json:"end_time"`
	Status          string    `gorm:"type:text" json:"status"`
	LocalBackupPath string    `json:"local_backup_path,omitempty"`
	TotalFiles      int       `json:"total_files"`
	TotalSizeBytes  int64     `json:"total_size_bytes"`
	ErrorMessage    string    `json:"error_message,omitempty"`
	Log             string    `json:"log,omitempty"`

	BackupFiles []BackupFile `json:"backup_files,omitempty"`
}
